2019-03-Autumn/6_dyn_prog/word_break/word_break.py

```diff
@@ -0,0 +1,26 @@
+def word_break(s: str, wordDict: list) -> bool:
+    wordDict = set(wordDict)
+    lengths = sorted(set(len(word) for word in wordDict))
+    dp = set()
+    dp.add(0)
+    for i in range(1, len(s) + 1):
```

Comment:
I agree that iterating over the string is the simplest solution but it could entail a fair amount of wasted work depending on the `wordDict` contents. This would be a good thing to address in a writeup.